nomos-services/network/src/backends/libp2p/mod.rs

```diff
@@ -41,24 +55,84 @@ impl NetworkBackend for Libp2p {
     type NetworkEvent = Event;
 
     fn new(config: Self::Settings, overwatch_handle: OverwatchHandle) -> Self {
-        let (commands_tx, commands_rx) = tokio::sync::mpsc::channel(BUFFER_SIZE);
+        let (swarm_commands_tx, swarm_commands_rx) = tokio::sync::mpsc::channel(BUFFER_SIZE);
         let (events_tx, _) = tokio::sync::broadcast::channel(BUFFER_SIZE);
 
-        let mut swarm_handler =
-            SwarmHandler::new(&config, commands_tx.clone(), commands_rx, events_tx.clone());
+        let mut swarm_handler = SwarmHandler::new(
+            &config,
+            swarm_commands_tx.clone(),
+            swarm_commands_rx,
+            events_tx.clone(),
+        );
+
+        // Run mixnode
+        let (mixnode, packet_queue) = MixNode::new(config.mixnode_config).unwrap();
+        let swarm_cmd_tx = swarm_commands_tx.clone();
+        let queue = packet_queue.clone();
+        overwatch_handle.runtime().spawn(async move {
+            Self::run_mixnode(mixnode, queue, swarm_cmd_tx).await;
+        });
+        let incoming_streams = swarm_handler.incoming_streams(STREAM_PROTOCOL);
+        let runtime_handle = overwatch_handle.runtime().clone();
+        let queue = packet_queue.clone();
+        overwatch_handle.runtime().spawn(async move {
+            Self::handle_incoming_streams(incoming_streams, queue, runtime_handle).await;
+        });
+
+        // Run mixclient
+        let (mixclient, message_queue) = MixClient::new(config.mixclient_config).unwrap();
+        let swarm_cmd_tx = swarm_commands_tx.clone();
+        overwatch_handle.runtime().spawn(async move {
+            Self::run_mixclient(mixclient, packet_queue, swarm_cmd_tx).await;
+        });
+
```

Comment:
I'm confused. Why is this overriding the old libp2p protocol instead of implementing a new one for the mixnet?

Reply:
Yeah, I think this is related to what you and I discussed in Discord (whether all nomos-node must enable mixnet). In the current spec, we expect all nomos-nodes enable mixnet. 

In the previous PR, I defined a new network backend for mixnet (`backends/mixnet/mod.rs`), but I removed it in this PR because I thought the "mixnet" is something that should be enabled as an extension for every network backend. We have only one network backend (libp2p) for now, but I imagined other network backends that we may have in the future. 

So, I made this libp2p network backend use the `mixnet` crate directly. The `mixnet` crate has only core mixnet functionalities without depending on any networking stuff (QUIC stream, and gossipsub). Also, this structure is what we've had previously (in the master branch).

What do you think about this?

Reply:
I think we should have kept the old libp2p backend, and implement a mixnet backend. Reason is that we may at some point test things without the mixnet itself, libp2p backend may still come in handy 🤔 
What I mean is that as they are replaceable pieces we should try to keep things that work, and just remove them when we decide we do not want to maintain them anymore.

Reply:
I also totally agree with this point. That's also why I tried define a new backend for mixnet :) 

Good. Then, I'll work on it. I'm not sure if it would be good to modify this PR or open a new PR. I'll first try and find the better way. But, the internal implementation wouldn't be so different from this PR. So, you can continue to review other existing PRs.

Reply:
Yeah, I'll continue with other PRs.
Up to you if making a new one or using this one :)

Reply:
I just pushed a new commit in this PR, which defines the `MixnetNetworkBackend`. I recommend to review the whole change of this PR, instead of looking at a new commit.